src/REPLMode/REPLMode.jl

```diff
@@ -604,6 +604,11 @@ function repl_init(repl::REPL.AbstractREPL)
         end
     )
     main_mode.keymap_dict = LineEdit.keymap_merge(main_mode.keymap_dict, keymap)
+    shell_mode_index = findfirst(x->startswith(x.prompt, REPL.SHELL_PROMPT), repl.interface.modes)
```

Comment:
It's either detecting the shell mode by its prompt, or relying on it being second in the list `repl.interface.modes`. I opted for the former.

Reply:
Relying on the name would not work with a customized prompt, e.g. [OhMyREPL](https://kristofferc.github.io/OhMyREPL.jl/latest/features/prompt/#Prompt-changing) provides methods for doing this for julia mode, but similar things would be easy to do for other modes.

Though maybe if one goes to the trouble of changing the prompt, it wouldn't be too much more effort to also update a keymap if it was desired. And it is not like it breaks, just that the map wouldn't be there. 

But it does seem a little brittle to rely on the name, seems like it would be nice to have a proper identifier for these things.